woocommerce/woocommerce-ios
# Migrate Observables to Combine: product form

Parts of #4379 


### Description


We used to have our own implementation of observables before we can use Combine in iOS 13, and now we're ready to migrate to Combine in the codebase. This PR only contains the changes for product form related files for easier review & testing.

During testing, I noticed an issue where the product images aren't updated in product form after uploading an image. This was due to the `AnyCancellable` subscription not being held onto and was canceled immediately in Combine, and fixed in https://github.com/woocommerce/woocommerce-ios/commit/bd5f046c7ac1e47fc5460cb0e16ea479c34a0ef0. Similarly, failing unit tests were fixed this way.

### Testing instructions


Please test for regressions on product editing:

- Launch the app
- Go to the products tab
- Tap on an editable product
- Try updating product images, name, description, etc. --> the "Update"/"Save" button should be shown only when there are actual changes on the product. everything should work as before in product editing
- Feel free to try editing the downloadable files for a downloadable product

---
